WB-1671: Dropdown: use `combobox` role in all openers (#2345)
Khan/wonder-blocks: .changeset/tasty-rockets-mix.md

---
"@khanacademy/wonder-blocks-dropdown": patch
---

> Update dropdown openers to use `role="combobox"` instead of `button`.

Reviewer: (very) nit: The ActionMenu also uses DropdownOpener and it still uses role="button"!

```suggestion
Update dropdown openers for SingleSelect and MultiSelect to use `role="combobox"` instead of `button`.
```

Reviewer 2: Seems very important to me! Good call!

Reviewer 2: Added another note to the changelog about labeling!

---
"@khanacademy/wonder-blocks-dropdown": major
---

1. Updates dropdown openers for SingleSelect and MultiSelect to use `role="combobox"` instead of `button`.

> 2. SingleSelect and MultiSelect now also require a paired `<label>` element or `aria-label` attribute for accessibility. They no longer fall back to text content for labeling, as those contents are now used as combobox values.

Reviewer: small nit: Might be worth rephrasing the "required" part a bit since we aren't changing the API to enforce this (aria-label is still optional and we don't have anything in the component checking if there is an associated label element). Maybe: 
```suggestion
2. SingleSelect and MultiSelect should have a paired `<label>` element or `aria-label` attribute for accessibility. They no longer fall back to text content for labeling, as those contents are now used as combobox values.
```
